### ISSUE TITLE
webapp fails to re-create config.xml when original is sym link to non-writable file

### ISSUE THREAD

**Issue by white-haired-uncle** [MythTV/mythtv#919]
* **Platform**:
`Linux mythbackend 6.5.0-35-generic #35~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Tue May  7 09:00:52 UTC 2 x86_64 x86_64 x86_64 GNU/Linux`

* **MythTV version**:
mythbackend$ mythbackend --version
Please attach all output as a file in bug reports.
MythTV Version : v34.0-18-g5590d0d5eb
MythTV Branch : fixes/34
Network Protocol : 91
Library API : 34.20220913-1
QT Version : 5.15.3
Options compiled in:
 linux profile use_hidesyms using_alsa using_oss using_pulse using_pulseoutput using_backend using_bindings_perl using_bindings_python using_bindings_php using_dvb using_firewire using_frontend using_hdhomerun using_satip using_vbox using_ceton using_joystick_menu using_libcec using_libcrypto using_gnutls using_libdns_sd using_libxml2 using_lirc using_mheg using_opengl using_egl using_qtwebkit using_qtscript using_qtdbus using_taglib using_v4l2 using_v4l2prime using_x11 using_system_libbluray using_system_libudfread using_systemd_notify using_systemd_journal using_drm using_bindings_perl using_bindings_python using_bindings_php using_freetype2 using_mythtranscode using_opengl using_egl using_drm using_vaapi using_nvdec using_vdpau using_ffmpeg_threads using_mheg using_libass using_libxml2 using_libmp3lame

* **Package version**:

* **Component**:
webapp
### What steps will reproduce the bug?

1. Replace ~mythtv/.mythtv/config.xml with a symlink to /etc/mythtv/config.xml (owned by root, not writable by mythtv)
2. Make a valid change at http://mythbackend:6544/setupwizard/dbsetup and save
3. Observe that ~mythtv/.mythtv/config.xml is now a regular but empty file, even though webapp said it saved successfully

### How often does it reproduce? Is there a required condition?

Always, as long as the target is not writable by mythtv

### What is the expected behaviour?

I would expect to see a permission denied error, and the link to remain in place so that mythbackend can be restarted without manual intervention.

**Comment by bennettpeter** [comment]
I tried this on latest master and I do not see the same thing you got. I moved config.xml to /etc/mythtv/ and set owner and group to root:root, created a symlink in .mythtv. I started mythbackend, I went to localhost:6544, went to setup , enable updates, change db server from localhost to 127.0.0.11, saved. Now the symlink config.xml in .mythtv is changed to a regular file, length 1150, with all of the correct data. The only difference between the file in /etc/mythtv and the one in .mythtv is the host ip address: one has localhost and the othet has 127.0.0.1. Restarting mythbackend is normal and database is accessed normally.

I think this is a better result than a permission denied error.

There must be some special condition which caused your problem.
